lib/manageiq/api/common/headers.rb

```diff
@@ -11,6 +12,11 @@ def self.current
           Thread.current[:attr_current_headers]
         end
 
+        def self.forwardable_headers
+          raise "Current Headers has not been set" unless Thread.current[:attr_current_headers]
+          Thread.current[:attr_current_headers].to_h.slice(*FORWARDABLE_HEADER_KEYS)
```

Comment:
Minor, but this should use the accessor instead of going directly to the Thread local...

```ruby
        def self.forwardable_headers
          raise "Current Headers has not been set" unless current
          current.to_h.slice(*FORWARDABLE_HEADER_KEYS)
        end
```